sqlglot/dialects/tsql.py

```diff
@@ -35,6 +68,20 @@ class Parser(Parser):
         FUNCTIONS = {
             **Parser.FUNCTIONS,
             "CHARINDEX": exp.StrPosition.from_arg_list,
+            "DATENAME": lambda args: exp.TimeToStr(
+                this=list_get(args, 1),
+                format=exp.Literal(
+                    this=format_time(list_get(args, 0).sql(), {**TSQL.time_mapping, **TSQL.date_name_mapping}),
```

Comment:
Hi,

First attempt was with the format_time_lambda method which I think is the preferred solution?

Stepped away from that because the format_time expectes a string. 
Because the parameters in TSQL for these functions are not wrapped in quotes the tokenizer produces a Column: Identifier for the parameter. 
So I went with the option to forcefully take the value as a string into the format_time. I'll try to see if I can come up with something better. Do you have any suggestions or tips?